pict-lib/pict/private/pict.rkt

```diff
@@ -792,14 +927,23 @@
                 cc-superimpose
                 ctl-superimpose
                 cbl-superimpose)
-  (let ([make-superimpose 
+  (let ([make-superimpose
          (lambda (get-h get-v get-th name)
-           (lambda boxes
-             (when (null? boxes)
+           (lambda boxes*
+             (when (null? boxes*)
                (error name "expected at least one argument, got none"))
-             (unless (andmap pict? boxes)
-               (error name "expected all picts as arguments, got ~a"
-                      (apply string-append (add-between (map (λ (x) (format "~e" x)) boxes) " "))))
+             (define boxes
+               (map
+                (lambda (p)
+                  (cond
+                    [(pict? p) p]
+                    [(pict-convertible? p)
+                     (pict-convert p)]
+                    [else
+                     (raise-argument-error
+                      name "all picts as arguments"
+                      (apply string-append (add-between (map (λ (x) (format "~e" x)) boxes*) " ")))]))
```

Comment:
It's written this way only because the original code was written this way and I did want to change the error message.

https://github.com/racket/pict/blob/master/pict-lib/pict/private/pict.rkt#L801